cdc/processor/pipeline/table_actor.go

```diff
@@ -0,0 +1,457 @@
+			// No need to handle remaining messages.
+			return false
+		}
+
+		var err error
+		switch msgs[i].Tp {
+		case message.TypeBarrier:
+			err = t.handleBarrierMsg(ctx, msgs[i].BarrierTs)
+		case message.TypeTick:
+			err = t.handleTickMsg(ctx)
+		case message.TypeStop:
+			t.handleStopMsg(ctx)
+		}
+		if err != nil {
+			log.Error("failed to process message, stop table actor ",
+				zap.String("tableName", t.tableName),
+				zap.Int64("tableID", t.tableID),
+				zap.Any("message", msgs[i]),
+				zap.Error(err))
+			break
+		}
+
+		// process message for each node, pull message from parent node and then send it to next node
+		if err := t.handleDataMsg(ctx); err != nil {
+			log.Error("failed to process message, stop table actor ",
+				zap.String("tableName", t.tableName),
+				zap.Int64("tableID", t.tableID), zap.Error(err))
+			t.handleError(err)
+			break
+		}
+	}
+	return atomic.LoadUint32(&t.stopped) != stopped
+}
+
+func (t *tableActor) handleDataMsg(ctx context.Context) error {
+	for _, n := range t.nodes {
+		if err := n.TryRun(ctx); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (t *tableActor) handleBarrierMsg(ctx context.Context, barrierTs model.Ts) error {
+	t.sortNode.UpdateBarrierTs(barrierTs)
+	return t.sinkNode.UpdateBarrierTs(ctx, barrierTs)
+}
+
+func (t *tableActor) handleTickMsg(ctx context.Context) error {
+	// tick message flush the raw event to sink, follow the old pipeline implementation, batch flush the events  every 500ms
+	if time.Since(t.lastFlushSinkTime) > sinkFlushInterval {
+		_, err := t.sinkNode.HandleMessage(ctx, pipeline.TickMessage())
+		if err != nil {
+			return err
+		}
+		t.lastFlushSinkTime = time.Now()
+	}
+	return nil
+}
+
+func (t *tableActor) handleStopMsg(ctx context.Context) {
+	// async stops sinkNode and tableSink
+	go func() {
+		_, err := t.sinkNode.HandleMessage(ctx,
+			pipeline.CommandMessage(&pipeline.Command{Tp: pipeline.CommandTypeStop}),
+		)
+		if err != nil {
+			t.handleError(err)
+		}
+	}()
+}
+
+func (t *tableActor) start(sdtTableContext context.Context) error {
+	if t.started {
+		log.Panic("start an already started table",
+			zap.String("changefeedID", t.changefeedID),
+			zap.Int64("tableID", t.tableID),
+			zap.String("tableName", t.tableName))
+	}
+	log.Debug("creating table flow controller",
+		zap.String("changefeedID", t.changefeedID),
+		zap.Int64("tableID", t.tableID),
+		zap.String("tableName", t.tableName),
+		zap.Uint64("quota", t.memoryQuota))
+
+	flowController := common.NewTableFlowController(t.memoryQuota)
+	sorterNode := newSorterNode(t.tableName, t.tableID,
+		t.replicaInfo.StartTs, flowController,
+		t.mounter, t.replicConfig,
+	)
+	sortActorNodeContext := NewContext(sdtTableContext, t.tableName,
+		t.globalVars.TableActorSystem.Router(),
+		t.actorID, t.changefeedVars, t.globalVars, t.reportErr)
+	if err := sorterNode.StartActorNode(sortActorNodeContext, true, t.wg, t.actorID, t.router); err != nil {
```

Comment:
May be it is better change this method name as `start` cause we already know it is belong to sorterNode and it is only called in this package.

Reply:
done